File: rb/features.rb
# Elonmuskapi SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ElonmuskapiFeatures
  def self.make_feature(name)
    case name
    when "base"
      ElonmuskapiBaseFeature.new
    when "ratelimit"
      ElonmuskapiRatelimitFeature.new
    when "retry"
      ElonmuskapiRetryFeature.new
    when "test"
      ElonmuskapiTestFeature.new
    when "timeout"
      ElonmuskapiTimeoutFeature.new
    else
      ElonmuskapiBaseFeature.new
    end
  end
end

File: py/features.py
# Elonmuskapi SDK feature factory

from feature.base_feature import ElonmuskapiBaseFeature
from feature.test_feature import ElonmuskapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ElonmuskapiBaseFeature(),
        "test": lambda: ElonmuskapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
